godot-core/src/obj/traits.rs

```diff
@@ -210,6 +210,22 @@ pub trait IndexEnum: EngineEnum {
 /// Trait that's implemented for user-defined classes that provide a `Base<T>` field.
 ///
 /// Gives direct access to the containing `Gd<Self>` from `Self`.
+///
+/// # Using WithBaseField as a bound
+///
+/// In order to call `self.base()` or `self.base_mut()` within a trait or on a type you define, the type of `Self::Base` must be specified via `GodotClass<Base = T>`
```

Comment:
Updating description to match https://github.com/godot-rust/gdext/pull/716#discussion_r1606069867

```suggestion
/// In order to call `self.base()` or `self.base_mut()` within a trait or on a type you define, the type of `Self::Base` must be specified via `WithBaseField<Base = T>`
```